chapter_recurrent-neural-networks/bi-rnn.md

```diff
@@ -6,67 +6,75 @@ So far we assumed that our goal is to model the next word given what we have see
 
 ## Dynamic Programming
 
-This section serves to *illustrate* the problem. The specific technical details do not matter for understanding the deep learning counterpart but they help in motivating why one might use deep learning and why one might pick specific architectures.
+This subsection serves to illustrate the dynamic programming problem. The specific technical details do not matter for understanding the deep learning counterpart but they help in motivating why one might use deep learning and why one might pick specific architectures.
 
-If we want to solve the problem using graphical models we could for instance design a latent variable model as follows: we assume that there exists some latent variable $h_t$ which governs the emissions $x_t$ that we observe via $p(x_t|h_t)$. Moreover, the transitions $h_t \to h_{t+1}$ are given by some state transition probability $p(h_t|h_{t-1})$. The graphical model then looks as follows:
+If we want to solve the problem using graphical models we could for instance design a latent variable model as following. We assume that there exists some latent variable $h_t$ which governs the emissions $x_t$ that we observe via $p(x_t|h_t)$. Moreover, the transitions $h_t \to h_{t+1}$ are given by some state transition probability $p(h_t+1|h_{t})$. The graphical model then looks as following:
 
 ![ Hidden Markov Model. ](../img/hmm.svg)
 
-For a sequence of $T$ observations we have thus the following joint probability distribution over observed and hidden states:
+Thus, for a sequence of $T$ observations we have the following joint probability distribution over the observed and the hidden states:
 
-$$p(x,h) = p(h_1) p(x_1|h_1) \prod_{i=2}^T p(h_t|h_{t-1}) p(x_t|h_t)$$
+$$p(x,h) = p(h_1) p(x_1|h_1) \prod_{t=2}^T p(h_t|h_{t-1}) p(x_t|h_t).$$
 
-Now assume that we observe all $x_i$ with the exception of some $x_j$ and it is our goal to compute $p(x_j|x^{-j})$. To accomplish this we need to sum over all possible choices of $h = (h_1, \ldots, h_T)$. In case $h_i$ can take on $k$ distinct values this means that we need to sum over $k^T$ terms - mission impossible! Fortunately there is an elegant solution for this: dynamic programming. To see how it works consider summing over the first two hidden variable $h_1$ and $h_2$. This yields:
+Now assume that we observe all $x_i$ with the exception of some $x_j$ and it is our goal to compute $p(x_j|x^{-j})$, where $x^{-j} = (x_1, x_2, \ldots, x_{j-1})$. To accomplish this we need to sum over all possible choices of $h = (h_1, \ldots, h_T)$. In case $h_i$ can take on $k$ distinct values, this means that we need to sum over $k^T$ terms - mission impossible! Fortunately there is an elegant solution for this: dynamic programming. To see how it works, consider summing over the first two hidden variable $h_1$ and $h_2$. This yields:
```

Comment:
the newly added "where $x^{-j} = (x_1, x_2, \ldots, x_{j-1})$"
This is a mistake because it should be $x^{-j} = (x_1, x_2, \ldots, x_{j-1}), x_{t+1}, \ldots, x_T$. 

I just removed it: https://github.com/d2l-ai/d2l-en/commit/ec6bbafb4d6ebdf937c75571150e9cb25355993e